Mathlib/LinearAlgebra/LinearIndependent/Defs.lean

```diff
@@ -160,6 +160,11 @@ theorem linearIndependent_zero_iff [Nontrivial R] : LinearIndependent R (0 : ι
 theorem linearIndepOn_zero_iff [Nontrivial R] : LinearIndepOn R (0 : ι → M) s ↔ s = ∅ :=
   linearIndependent_zero_iff.trans isEmpty_coe_sort
 
+@[simp]
+theorem linearIndependent_subsingleton_iff [Nontrivial R] [Subsingleton M] (f : ι → M) :
+    LinearIndependent R f ↔ IsEmpty ι := by
+  rw [Subsingleton.elim f 0, linearIndependent_zero_iff]
```

Comment:
I don't understand this move as well. This is a lemma and a def. 
Indeed there are confusingly already some lemmas in the def file but I'm not sure if we should make things worse.

Reply:
I'm moving this right below the lemma `linearIndependent_zero_iff` that it depends on; I think it improves coherence.
This file is full of lemmas so this is totally normal. It has 750 lines and might be split some time in the future.